File: step34_event-driven-restaurant-app/Python/cdk-backend/lambda-fns/snsHandler.py
import boto3
import json
import os
import logging
from botocore.exceptions import ClientError
sns = boto3.client('sns')

def handler(event, context):  
    print(event)
    if event["operationSuccessful"] == False:
        return json.dumps({ "message": "operation not successful" })
 

    try:
        if event["SnsMessage"]:
            #  sending message to TOPIC ARN
            sns.publish(
                TopicArn= os.environ["SNS_TOPIC_ARN"],
                Message= event["SnsMessage"],
            )
            print('message published')

            #  sending message to Phone Number
            sns.publish(
                Message= event["SnsMessage"],
                PhoneNumber = "+920020394066"
                # PhoneNumber=os.environ["PHONE_NUMBER"],
            )
            print('message sent to Phone.no:', os.environ["PHONE_NUMBER'})
    
    except ClientError as e:
        logging.error(e)
        print("error", e)
        print("ERROR Publishing To SNS ")

    return json.dumps({ "message": "operation successful" })

File: step34_event-driven-restaurant-app/cdk-backend/lambda-fns/snsHandler.ts
import { Context } from 'aws-lambda';
import { SNS,SES } from 'aws-sdk';
import { PayloadType } from './dynamoHandler';

const sns = new SNS();
var ses = new SES({ region: process.env.OUR_REGION });

export const handler = async (event: PayloadType, context: Context) => {
    console.log("event",event);

    // if (!event.operationSuccessful) {
    //     return { message: "operation not successful" }
    // }

    try {

        if (event.SnsMessage && !event.customerEmail) {
            // sending message to TOPIC ARN
            await sns.publish({
                TopicArn: process.env.SNS_TOPIC_ARN,
                Message: event.SnsMessage,
            }).promise()
            console.log('message published');

            // sending message to Phone Number
            await sns.publish({
                Message: event.SnsMessage,
                PhoneNumber: process.env.PHONE_NUMBER,
            }).promise()
            console.log('message sent to Phone.no:', process.env.PHONE_NUMBER);

        }
    }
    catch (err) {
        console.error('ERROR Publishing To SNS ====>', JSON.stringify(err, null, 2));
        throw new Error(err.message)
    }

    return { message: "operation successful" }

}